### PR TITLE
First generation documentation overhaul

### PR DESCRIPTION
**Pull request by camertron**
There's lots more to do, but this is a first attempt at a README overhaul. Document all the things.

@lumoslabs/platform

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,80 +1,229 @@
...
 
-You configure a service hook in Github and point it to this server. The URL path to the service hook endpoint: /hooks/github
-You do the same for Transifex, in your project settings page, and point it to the service hook endpoint: /hooks/transifex
+Txgh is distributed as a [Docker image](https://quay.io/repository/lumoslabs/txgh) and as a [Rubygem](https://rubygems.org/gems/txgh). You can choose to run it via Docker, install and run it as a Rubygem, or run it straight from a local clone of this repository.
 
-Currently there are 4 use cases that are supported:
+### With Docker
 
-1) When a resource (configured in this service) in Transifex reaches 100% translated, the Txgh service will pull the translations and commit them to the target repository.
+Using Docker to run Txgh is pretty straightforward (keep in mind you'll need to have the Docker client set up wherever you want to run Txgh).
```
Comment (jdoconnor) [inline README.md]: docker server. the client is the CLI or API that tells the server to do stuff like run or pull

File: README.md [hunk 2]
```diff
@@ -1,80 +1,229 @@
...
 
-How run it
----
+```bash
+curl -v localhost:9292/health_check
+....
+< HTTP/1.1 200 OK
+```
+
+Note that Txgh might not be available on localhost depending on how your Docker client is configured. On a Mac with [docker-machine](https://docs.docker.com/machine/) for instance, you might try this instead:
```
Comment (jdoconnor) [inline README.md:147]: append 
`where 192.168.99.100 is the IP of your docker machine instance`

**Comment by jdoconnor** [comment]
+1 with a few comments. Have you considered putting the defailt work flow up top so that someone can get started quickly and leave the options to be called out in lower sections? I'm not sure if that would be better or not.

**Comment by camertron** [comment]
@jdoconnor yes, that's coming up.

**Pull request merged**